### PR TITLE
Do not allow cleanup actions to run between builds with warm daemon

### PR DESCRIPTION
**Pull request by lptr**
Warm daemons have a lot of in-memory state that could get confused if these directories are deleted while they are alive.

### CODE REVIEW

File: src/main/java/org/gradle/profiler/GradleBuildInvoker.java [hunk 1]
```diff
@@ -19,6 +19,11 @@ private GradleBuildInvoker(GradleClientSpec client, GradleDaemonReuse daemonReus
         this.daemonReuse = daemonReuse;
     }
 
+    @Override
+    public boolean allowsCleanupBetweenBuilds() {
+        return daemonReuse == GradleDaemonReuse.ColdDaemonOnly;
```
Comment (wolfs) [inline src/main/java/org/gradle/profiler/GradleBuildInvoker.java:24]:
```suggestion
        return !isReuseDaemon();
```

**Review by wolfs — CHANGES_REQUESTED** [review]
I wonder if we are going to have a problem when integrating the changes about instantiating the BuildMutators directly. I suppose we shouldn't, right?

File: src/main/java/org/gradle/profiler/GradleBuildInvoker.java (hunk 1 above)
Comment (lptr) [reply src/main/java/org/gradle/profiler/GradleBuildInvoker.java]: Added with a comment.

**Comment by lptr** [comment]
There will be some changes necessary, but I think it's going to be fine on the Gradle side (and considerably simpler, too).

**Review by wolfs — APPROVED** [review]
LGTM!

**Pull request merged**